File: leetcode/Easy/Maximize-Sum-Of-Array-After-K-Negations.cpp
1class Solution {
2public:
3    int largestSumAfterKNegations(vector<int>& nums, int k) {
4        sort(nums.begin(),nums.end());
5        for(int &num : nums)
6        {
7            if(num < 0 && k)
8            {
9                num = -num;
10                --k;
11            }
12        }
13        sort(nums.begin(),nums.end());
14        int sum{0};
15                for(int &num : nums)
16                {
17                    sum+=num;
18                }
19            
20                if(k && k % 2 != 0)
21                    sum-=(2*nums[0]);
22
23            return sum; 
24
25
26    }
27};

File: leetcode/Easy/Maximize-Sum-Of-Array-After-K-Negations.java
1class Solution {
2    public int largestSumAfterKNegations(int[] nums, int k) {
3        Arrays.sort(nums);
4        int  mn = Integer.MAX_VALUE;
5        int sum = 0;
6        for(int i = 0 ; i < nums.length;++i)
7        {
8            if(nums[i] < 0 && k > 0)
9            {
10                nums[i]*=-1;
11                --k;
12            }
13            sum+=nums[i];
14            mn = Math.min(mn,nums[i]);
15        }
16        if(k % 2 != 0)
17            sum-=(2*mn);
18        return sum;
19        
20        
21    }
22}